R/preprocessing.R

```diff
@@ -9,9 +9,9 @@ NULL
 #' @param data dataset to be provided, default read_data() from PRIMACE github repo by read_data() function
 #' @return uploaded preprocessed dataset
 #' @export
-preprocessing <- function(train=TRUE, data=read_data()){
+preprocessing <- function(train=TRUE, datasets=read_data_s3()){
   bmr <- tab <- NULL
-  validate(need(is.data.frame(data), "No valid dataset is provided"))
+  data <- datasets$train
```

Comment:
In `tryCatch` function you must add a `return(one_year_dataset)` to expose this variable for next steps

Reply:
Yes, because `error` part in contrary to `expr` part returns `NA` value in case of error.

Reply:
assing `one_year_dataset` to `tryCatch` as:
```r
one_year_dataset <- tryCatch(

)
```
not defining variable inside `tryCatch`

Reply:
ok